server/utils/passport/ensure-authenticated.js

```diff
@@ -0,0 +1,44 @@
+import basePathConfig from '../../configs/basePathConfig'
+
+const basePath = basePathConfig || ''
+
+export default function ensureAuthenticated(req, res, next) {
+  if (!req.isAuthenticated()) {
+
+    return res.redirect(`${basePath}/logout`)
+  }
+  const appDb = req.app.locals.appDb
+  appDb
+    .collection('users')
+    .findOne(
+      { uid: req.user },
+      { 
+        _id: 0, 
+        access: 1, 
+        blocked: 1, 
+        role: 1 
+      },
+      function (err, data) {
+        if (err) {
+          console.log(err);
+        
+          return res.redirect(`${basePath}/logout?e=forbidden`)
+        } else if (!data || Object.keys(data).length === 0) {
+        
+          return res.redirect(`${basePath}/logout?e=forbidden`)
+        } else if (('role' in data) && data['role'] === 'admin') {
+        
+          return next();
+        } else if (('blocked' in data) && data['blocked'] == true) {
+        
+          return res.redirect(`${basePath}/logout?e=forbidden`)
+        } else if (!('access' in data) || data.access.length == 0) {
+        
+          return res.redirect(`${basePath}/logout?e=unauthorized`)
+        } else {
+        
+          return next();
+        }
```

Comment:
This might be a good use case for the `switch(true)` pattern

```suggestion
        } 

        switch(true) {
          case !data || Object.keys(data).length === 0 || !!data.blocked:
            return res.redirect(`${basePath}/logout?e=forbidden`)
          case data.access && data.access.length === 0:
            return res.redirect(`${basePath}/logout?e=unauthorized`)
          default:
            return next()
        }
```

Reply:
[606bdb4](https://github.com/AMP-SCZ/dpdash/pull/83/commits/606bdb41cd3775b9924079ea5bae6cf32b3230bb)